frontend/public/components/utils/details-item.tsx

```diff
@@ -47,36 +59,59 @@ export const DetailsItem: React.FC<DetailsItemProps> = ({
   const value: React.ReactNode = children || _.get(obj, path, defaultValue);
   return (
     <>
-      <dt>
-        {description ? (
-          <Popover
-            headerContent={<div>{label}</div>}
-            bodyContent={
-              <LinkifyExternal>
-                <div className="co-pre-line">{description}</div>
-              </LinkifyExternal>
-            }
-            footerContent={<PropertyPath kind={model.kind} path={path} />}
-            maxWidth="30rem"
-          >
-            <Button variant="plain" className="co-m-pane__details-popover-button">
-              {label}
-            </Button>
-          </Popover>
-        ) : (
-          label
-        )}
+      <dt className={classnames('details-item__label', labelClassName)}>
+        <Split>
+          <SplitItem className="details-item__label">
+            {description ? (
+              <Popover
+                headerContent={<div>{label}</div>}
+                bodyContent={
+                  <LinkifyExternal>
+                    <div className="co-pre-line">{description}</div>
+                  </LinkifyExternal>
+                }
+                footerContent={<PropertyPath kind={model.kind} path={path} />}
+                maxWidth="30rem"
+              >
+                <Button variant="plain" className="details-item__popover-button">
+                  {label}
+                </Button>
+              </Popover>
+            ) : (
+              label
+            )}
+          </SplitItem>
+          {editModal && (
```

Comment:
We might want the component to support the both edit styles, the inline edit links like annotations and groups. The styling could be controlled by a prop. Maybe `onEdit: (e) => void` and `editGroup: boolean`? Then if we ever need to change it, we can change it in one place. (I'm open to better suggestions for the prop names.)

Then we can slowly switch the existing modal links over to `DetailsItem`.

Reply:
Fixed